### DISCUSSION TITLE
希望可以在”项目“页面添加分组功能

### DISCUSSION THREAD

**Discussion by L0st1** [1111mp/nvm-desktop#76] [Ideas]
分组可以作为一个分类依据或者标签，用于批量指定版本
当项目可以存放在分组后，可以指定分组的版本，指定后分组中的每个项目都使用该版本
![屏幕截图 2024-04-10 142208](https://github.com/1111mp/nvm-desktop/assets/64479959/59166179-d8cd-48c7-80d5-51ab6afad955)

如图，有些项目使用的是14.17.0，有些使用的是16+，如果可以批量指定，用户体验很大提升

**Comment by 1111mp** [comment]
可以的 项目多了管理起来确实会比较麻烦

我目前的想法是会新增一个 "分组管理" 的界面 

新增分组：
- 提供用户选择“node的版本”
- 勾选已添加过的项目（多选）

当然也提供编辑的功能：修改名称、node版本、以及关联的项目...

然后在项目界面新增一列：所属分组，当然也会提供切换分组的功能。不过为了避免和现有的“node版本选择”的列在功能上冲突，目前想法是将“分组选择”和“node版本选择”放在一个“select”下（不同的分组），用户直接选择“node版本”时直接使用该版本，选择“分组名”时使用该分组设置的版本。

你看这种方式是否符合你的预期。

当然如果你有更好的想法也欢迎一起交流一下，谢谢🙏

**Reply by L0st1** [reply]
感谢阅读和回复，首次向开源项目提功能建议也是诚惶诚恐。
以上完全可以满足我想象中的“分组”功能；
交互上感觉可以把“分组选择”和“node版本选择”放在一个表格单元格里，做成单选，默认“node版本选择”，这样不会影响现有功能

**Reply by 1111mp** [reply]
客气了，后续我会实现这个功能的，只不过目前在忙着其他事情，可能不会这么快，还望理解。

**Comment by 1111mp** [comment]
目前的 [Release v3.3.0](https://github.com/1111mp/nvm-desktop/releases/tag/v3.3.0) 中已经支持了这个功能 你可以升级到最新版本试用一下 期间发现任何bug还望随时沟通 谢谢

那么我就先把这个讨论关闭了 当然后续如果有任何问题都可以随时重新打开 感谢 🌹
